# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/base/nsAccUtils.cpp
# Commit: d07307df1220
# Full Hash: d07307df122035ff6ea8c68e3ccdf895d46cc2fd
# Author: James Teh <jteh@mozilla.com>
# Date: 2022-11-10 09:23:09
# Regressor Bug: 1796733
# File Overlap Count: 1
# Description:
#   Bug 1796733 part 4: Unify nsAccUtils::GetSelectableContainer. r=morgan
#   
#   We need this to unify support for the implicit selected state on ARIA items.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D160881
# ==============================================================================

diff -r e8d222e1fd87 -r d07307df1220 accessible/base/nsAccUtils.cpp
--- a/accessible/base/nsAccUtils.cpp	Thu Nov 10 04:19:27 2022 +0000
+++ b/accessible/base/nsAccUtils.cpp	Thu Nov 10 04:19:27 2022 +0000
@@ -176,17 +176,24 @@
   return nullptr;
 }
 
-LocalAccessible* nsAccUtils::GetSelectableContainer(
-    LocalAccessible* aAccessible, uint64_t aState) {
+Accessible* nsAccUtils::GetSelectableContainer(const Accessible* aAccessible,
+                                               uint64_t aState) {
   if (!aAccessible) return nullptr;
 
   if (!(aState & states::SELECTABLE)) return nullptr;
 
-  LocalAccessible* parent = aAccessible;
-  while ((parent = parent->LocalParent()) && !parent->IsSelect()) {
+  const Accessible* parent = aAccessible;
+  while ((parent = parent->Parent()) && !parent->IsSelect()) {
     if (parent->Role() == roles::PANE) return nullptr;
   }
-  return parent;
+  return const_cast<Accessible*>(parent);
+}
+
+LocalAccessible* nsAccUtils::GetSelectableContainer(
+    LocalAccessible* aAccessible, uint64_t aState) {
+  Accessible* selectable =
+      GetSelectableContainer(static_cast<Accessible*>(aAccessible), aState);
+  return selectable ? selectable->AsLocal() : nullptr;
 }
 
 bool nsAccUtils::IsDOMAttrTrue(const LocalAccessible* aAccessible,